ui/js/components/auth.js

```diff
@@ -0,0 +1,165 @@
+             * for the tc-login, which will get closed if it's successful.
+             */
+            ctrl.login = function () {
+                var hash = encodeURIComponent("#");
+                var colon = encodeURIComponent(":");
+                var target = `${$location.protocol()}${colon}//${$location.host()}${colon}${$location.port()}/${hash}/login`;
+                var description = "Treeherder";
+                var url = `https://login.taskcluster.net/?target=${target}&description=${description}`;
+
+                // open a new tab to show the taskcluster auth login page
+                $window.open(url, "_blank");
+            };
+
+            /**
+             * Contact Treeherder back-end to log the user out and invalidate
+             * the session token.  Then updates the UI
+             */
+            ctrl.logout = function () {
+                $http.get(thUrl.getRootUrl("/auth/logout/"));
+                ctrl.setLoggedOut();
+            };
+
+            ctrl.setLoggedIn = function(newUser) {
+                newUser.loggedin = true;
+                localStorageService.set("user", newUser);
+                _.extend(ctrl.user, newUser);
+                ctrl.onUserChange({$event: {user: ctrl.user}});
+
+            };
+
+            ctrl.setLoggedOut = function() {
+                localStorageService.set("user", loggedOutUser);
+                _.extend(ctrl.user, loggedOutUser);
+                ctrl.onUserChange({$event: {user: loggedOutUser}});
+            };
+        }]
+});
+
+treeherder.component("loginCallback", {
+    template: `
+        <div ng-if="!loginError">
+            <span>Logging in... <span class="fa fa-spinner fa-spin"></span></span>
+        </div>
+        <div ng-if="loginError">
+            <span>Error logging in: {{loginError}}</span>
+        </div>
+    `,
+    controller: ['localStorageService', '$location', '$window', '$http', '$scope',
+        function(localStorageService, $location, $window, $http, $scope) {
+            const host = $location.host();
+            const port = $location.port();
+            const urlBase = `${$location.protocol()}://${host}:${port}/`;
+            const certificate = $location.search().certificate;
+            const credentials = {
+                id: $location.search().clientId,
+                key: $location.search().accessToken,
+                algorithm: 'sha256'
+            };
+            this.loginError = null;
+            const header = hawk.client.header(urlBase, 'GET', {
+                credentials: credentials,
+                ext: hawk.utils.base64urlEncode(JSON.stringify({"certificate": JSON.parse(certificate)}))}
+            );
+
+            // send a request from client side to TH server signed with TC
+            // creds from login.taskcluster.net
+            $http.get(`${urlBase}api/auth/login/`,
+                      {headers: {"oth": header.field}})
```

Comment:
Yeah, the "Authorization" header was what I'd tried at first, and you're exactly right.  It got confused with the Hawk middleware.  But, sure, I'll make this a POST instead.  Makes sense.